src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -0,0 +1,459 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.dependencytrack.tasks.scanners;
+
+import alpine.common.logging.Logger;
+import alpine.common.util.Pageable;
+import alpine.event.framework.Event;
+import alpine.event.framework.Subscriber;
+import alpine.model.ConfigProperty;
+import alpine.security.crypto.DataEncryption;
+import com.github.packageurl.MalformedPackageURLException;
+import com.github.packageurl.PackageURL;
+import kong.unirest.*;
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.dependencytrack.common.UnirestFactory;
+import org.dependencytrack.event.IndexEvent;
+import org.dependencytrack.event.SnykAnalysisEvent;
+import org.dependencytrack.model.*;
+import org.dependencytrack.parser.common.resolver.CweResolver;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.util.NotificationUtil;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.dependencytrack.model.ConfigPropertyConstants.SCANNER_SNYK_BASE_URL;
+import static org.dependencytrack.util.JsonUtil.jsonStringToTimestamp;
+
+/**
+ * Subscriber task that performs an analysis of component using Snyk vulnerability REST API.
+ */
+public class SnykAnalysisTask extends BaseComponentAnalyzerTask implements Subscriber {
+
+    private String ORG_ID;
+    private String API_BASE_URL;
+    private String BASE_URL;
```

Comment:
`BASE_URL` and `ORG_ID` should be local variables, as they're only ever used in `inform`.

Reply:
done